modules/instance_template/main.tf

```
@@ -71,13 +71,13 @@ resource "google_compute_instance_template" "tpl" {
       boot         = lookup(disk.value, "boot", null)
       device_name  = lookup(disk.value, "device_name", null)
       disk_name    = lookup(disk.value, "disk_name", null)
-      disk_size_gb = lookup(disk.value, "disk_size_gb", null)
+      disk_size_gb = lookup(disk.value, "disk_type", null) == "local-ssd" ? "375" : lookup(disk.value, "disk_size_gb", null)
       disk_type    = lookup(disk.value, "disk_type", null)
-      interface    = lookup(disk.value, "interface", null)
+      interface    = lookup(disk.value, "disk_type", null) == "local-ssd" ? var.default_local_ssd_interface : lookup(disk.value, "interface", null)
       mode         = lookup(disk.value, "mode", null)
       source       = lookup(disk.value, "source", null)
       source_image = lookup(disk.value, "source_image", null)
-      type         = lookup(disk.value, "type", null)
+      type         = lookup(disk.value, "disk_type", null) == "local-ssd" ? "SCRATCH" : lookup(disk.value, "type", null)
```

Shouldn't we always use the explicit `disk.type` value if provided?

---

The Terraform provider expects `disk_type` to be `local-ssd` and type to be `SCRATCH`.

Reference: https://github.com/hashicorp/terraform-provider-google/blob/master/google/resource_compute_instance_template.go#L670

And unfortunately, `type` cannot be set in `var.additional_disks`.

---

I see, couldn't we reverse this logic though?

```suggestion
       type         = lookup(disk.value, "type", lookup(disk.value, "disk_type", null) == "local-ssd" == "local-ssd" ? "SCRATCH" : null))     
```

---

Submitted the change for the logic reverse.